.github/workflows/buf-push-checks.yaml

```diff
@@ -0,0 +1,32 @@
+name: Buf push checks
+on:
+  push:
+    branches:
+      - main
```

Comment:
I think it will be better to have a single workflow file for all pushes.

Needing to hard code `main` here isn't great. In a unified workflow we can use `${{ github.event.repository.default_branch }}` to detect if we are on the right branch and behave accordingly.

Reply:
Starter workflows have `$default-branch` templating: https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization#creating-a-starter-workflow
And other workflow files will specify the default as part of their config. I don't think it's possible to access the github event as part of the specification of what events it listens on as the definition would be circular.

Reply:
Is buf-action using starter workflows?

> I don't think it's possible to access the github event as part of the specification of what events it listens on as the definition would be circular.

No you can't, but you can use `if:` statements for the jobs or steps.
https://stackoverflow.com/questions/64781462/github-actions-default-branch-variable

Reply:
Not currently, but it's something we could provide in addition to reusable workflows. I could see the use case where a user  wants to edit the pre-defined workflow and this would provide an eject mechanism.